x-pack/legacy/plugins/maps/public/layers/fields/es_doc_field.js

```diff
@@ -27,4 +27,31 @@ export class ESDocField extends AbstractField {
     return field.type;
   }
 
+  supportsFieldMeta() {
+    return true;
```

Comment:
imho, we should limit to only number-fields. (and later on `string` fields in subsequent PR).

Write this in terms of `this.getDataType() === 'number'`. This would require this method to become `async` though. I don't think this would make things too much more complicated (?) since `syncSource` is async, and rendering the legend is just UX where the asynciness is wrapped by React-component.

Reply:
Making `supportsFieldMeta` async will be problematic since its used in VectorStyle._getFieldRange which gets called by VectorStyle.setFeatureState which gets called by VectorLayer.syncLayerWithMB and making syncLayerWithMB async will be a problem